src/coreclr/jit/importer.cpp

```diff
@@ -11776,12 +11782,35 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     setMethodHasPatchpoint();
                 }
             }
+            else
+            {
+                // Should not see backward branch targets w/o backwards branches
+                assert((block->bbFlags & BBF_BACKWARD_JUMP_TARGET) == 0);
+            }
         }
-    }
-    else
-    {
-        // Should not see backward branch targets w/o backwards branches
-        assert((block->bbFlags & BBF_BACKWARD_JUMP_TARGET) == 0);
+
+#ifdef DEBUG
+        // As a stress test, we can place patchpoints at the start of any block
+        // that is a stack empty point and is not within a handler.
+        // Todo: enable for mid-block stack empty points too.
+        //
+        const int doRandomOSR = JitConfig.JitRandomOnStackReplacement();
+        if ((doRandomOSR > 0) && (verCurrentState.esStackDepth == 0) && !block->hasHndIndex() &&
+            ((block->bbFlags & BBF_PATCHPOINT) == 0))
+        {
+            // Reuse the random inliner's random state.
+            // Note m_inlineStrategy is always created, even if we're not inlining.
+            //
+            CLRRandom* const random      = impInlineRoot()->m_inlineStrategy->GetRandom(doRandomOSR);
+            const int        randomValue = (int)random->Next(101);
```

Comment:
Shouldn't this be `random->Next(100)` for `0x65` to be 100% probability?

Reply:
Yes.